sources/hubspot/__init__.py

```diff
@@ -38,54 +38,174 @@
+        )
+
+    @dlt.resource(name="contacts", write_disposition="replace")
+    def contacts(
+        api_key: str = api_key,
+        include_history: bool = include_history,
+        props: Sequence[str] = DEFAULT_CONTACT_PROPS,
+    ) -> Iterator[TDataItems]:
+        """Hubspot contacts resource"""
+        yield from crm_objects(
+            "contact",
+            api_key,
+            include_history,
+            props if props in (ALL, CUSTOM_ONLY) else props + global_props,  # type: ignore
+        )
+
+    @dlt.resource(name="deals", write_disposition="replace")
+    def deals(
+        api_key: str = api_key,
+        include_history: bool = include_history,
+        props: Sequence[str] = DEFAULT_DEAL_PROPS,
+    ) -> Iterator[TDataItems]:
+        """Hubspot deals resource"""
+        yield from crm_objects(
+            "deal",
+            api_key,
+            include_history,
+            props if props in (ALL, CUSTOM_ONLY) else props + global_props,  # type: ignore
+        )
+
+    @dlt.resource(name="tickets", write_disposition="replace")
+    def tickets(
+        api_key: str = api_key,
+        include_history: bool = include_history,
+        props: Sequence[str] = DEFAULT_TICKET_PROPS,
+    ) -> Iterator[TDataItems]:
+        """Hubspot tickets resource"""
+        yield from crm_objects(
+            "ticket",
+            api_key,
+            include_history,
+            props if props in (ALL, CUSTOM_ONLY) else props + global_props,  # type: ignore
+        )
+
+    @dlt.resource(name="products", write_disposition="replace")
+    def products(
+        api_key: str = api_key,
+        include_history: bool = include_history,
+        props: Sequence[str] = DEFAULT_PRODUCT_PROPS,
+    ) -> Iterator[TDataItems]:
+        """Hubspot products resource"""
+        yield from crm_objects(
+            "product",
+            api_key,
+            include_history,
+            props if props in (ALL, CUSTOM_ONLY) else props + global_props,  # type: ignore
+        )
+
+    @dlt.resource(name="quotes", write_disposition="replace")
+    def quotes(
+        api_key: str = api_key,
+        include_history: bool = include_history,
+        props: Sequence[str] = DEFAULT_QUOTE_PROPS,
+    ) -> Iterator[TDataItems]:
+        """Hubspot quotes resource"""
+        yield from crm_objects(
+            "quote",
+            api_key,
+            include_history,
+            props if props in (ALL, CUSTOM_ONLY) else props + global_props,  # type: ignore
+        )
+
+    return companies, contacts, deals, tickets, products, quotes
 
 
 def crm_objects(
     object_type: str,
     api_key: str = dlt.secrets.value,
     include_history: bool = False,
+    props: Sequence[str] = None,
 ) -> Iterator[TDataItems]:
     """Building blocks for CRM resources."""
-    props = ",".join(_get_property_names(api_key, object_type))
+    if props == ALL:
```

Comment:
shouldn't all be a tuple? `("ALL",)`. props are a sequence? also inter should complain here

Reply:
Well, actually, I against it. That's a strange value `("ALL",)`, like there is a property `ALL`. Users don't have to think about constants values, that's the idea of a constant - its name says everything the user need to now. Anyway, it's a tiny thing. Fixed.

Reply:
Heh I was not clear here. I suggested exactly what you did: keep the constant but the constant was not a tuple in `settings` which you fixed.